src/clj/puppetlabs/services/master/master_service.clj

```diff
@@ -4,25 +4,30 @@
             [puppetlabs.trapperkeeper.core :refer [defservice]]
             [puppetlabs.services.master.master-core :as core]
             [puppetlabs.puppetserver.certificate-authority :as ca]
-            [puppetlabs.trapperkeeper.services :as tk-services]))
+            [puppetlabs.trapperkeeper.services :as tk-services]
+            [puppetlabs.dujour.version-check :as version-check]))
 
 (defservice master-service
   [[:WebroutingService add-ring-handler get-route]
-   [:PuppetServerConfigService get-config]
+   [:PuppetServerConfigService get-config get-in-config]
    [:RequestHandlerService handle-request]
    [:CaService initialize-master-ssl! retrieve-ca-cert! retrieve-ca-crl!]
    [:JRubyPuppetService]]
   (init
    [this context]
    (core/validate-memory-requirements!)
-   (let [path          (get-route this :master-routes)
-         config        (get-config)
-         certname      (get-in config [:puppet-server :certname])
-         localcacert   (get-in config [:puppet-server :localcacert])
-         hostcrl       (get-in config [:puppet-server :hostcrl])
-         settings      (ca/config->master-settings config)
-         jruby-service (tk-services/get-service this :JRubyPuppetService)
-         upgrade-error (core/construct-404-error-message jruby-service)]
+   (let [path              (get-route this :master-routes)
+         config            (get-config)
+         certname          (get-in config [:puppet-server :certname])
+         localcacert       (get-in config [:puppet-server :localcacert])
+         hostcrl           (get-in config [:puppet-server :hostcrl])
+         settings          (ca/config->master-settings config)
+         jruby-service     (tk-services/get-service this :JRubyPuppetService)
+         upgrade-error     (core/construct-404-error-message jruby-service)
+         product-name      (get-in-config [:product :name])
```

Comment:
Do we need the `get-in-config`? We're already pulling in the whole `config` and using the standard `(get-in config [:k1 :k2])` in this let block